CSharpTemplate.cs

```diff
@@ -8,14 +8,13 @@ namespace CommBank.Api.Provisioning.Services.Source.Templates
 {
     public class CSharpTemplate : ISourceTemplate
     {
-        private static readonly string[] DefaultIgnoreDirs = { ".git", "node_modules", "bower_components", "packages", "bin", "obj" };
-
         private readonly string _name;
         private readonly string _repositoryUrl;
         private readonly string _solutionName;
         private readonly string _branchName;
         private readonly IDictionary<string, string> _templatedValues;
         private readonly string[] _ignoreDirs;
+        private readonly ITemplateIgnoreContentProvider _templateIgnoreContentProvider = new TemplateIgnoreContentProvider(new SlipstreamIgnoreFileLoader());
```

Comment:
move to another abstract class, and get the dependentcy injection from some third part abstract class